data-prepper-plugins/newline-codecs/src/main/java/org/opensearch/dataprepper/plugins/codec/newline/NewlineDelimitedOutputCodec.java

```diff
@@ -4,42 +4,80 @@
  */
 package org.opensearch.dataprepper.plugins.codec.newline;
 
+import com.fasterxml.jackson.databind.ObjectMapper;
 import org.opensearch.dataprepper.model.annotations.DataPrepperPlugin;
+import org.opensearch.dataprepper.model.annotations.DataPrepperPluginConstructor;
 import org.opensearch.dataprepper.model.codec.OutputCodec;
 import org.opensearch.dataprepper.model.event.Event;
 
 import java.io.IOException;
 import java.io.OutputStream;
+import java.util.Map;
+import java.util.Objects;
 
 /**
  * An implementation of {@link OutputCodec} which deserializes Data-Prepper events
  * and writes them to Output Stream as ND-JSON data
  */
-@DataPrepperPlugin(name = "newline", pluginType = OutputCodec.class)
+@DataPrepperPlugin(name = "newline", pluginType = OutputCodec.class, pluginConfigurationType = NewlineDelimitedOutputConfig.class)
 public class NewlineDelimitedOutputCodec implements OutputCodec {
+    private static final String NDJSON = "ndjson";
+    private static final String MESSAGE_FIELD_NAME = "message";
+    private static final ObjectMapper objectMapper = new ObjectMapper();
+    private final NewlineDelimitedOutputConfig config;
+
+    @DataPrepperPluginConstructor
+    public NewlineDelimitedOutputCodec(final NewlineDelimitedOutputConfig config) {
+        Objects.requireNonNull(config);
+        this.config = config;
+    }
 
     @Override
     public void start(final OutputStream outputStream) throws IOException {
-        // TODO: implement
+        Objects.requireNonNull(outputStream);
     }
 
     @Override
     public void writeEvent(final Event event, final OutputStream outputStream) throws IOException {
-        // TODO: get the event data and
-        //  get the header record and message record and write event data to the outputstream
+        Objects.requireNonNull(event);
+        final Map<String, Object> eventDataMap = event.toMap();
+        if (eventDataMap.containsKey(MESSAGE_FIELD_NAME)) {
```

Comment:
This still doesn't look correct. You don't need this if clause. You just need the line in the else clause. The NewLineInputCodec has the notion of MESSAGE_FIELD_NAME, but you can assume at the sink that the entire event (minus exclude_keys) should get written to the sink.

Reply:
@asuresh8  updated the code.